### ISSUE TITLE
Control flow analysis gives false positive 'Use of moved value' when noreturn function used

### ISSUE THREAD

**Issue by MikailBag** [intellij-rust/intellij-rust#3251]
## Environment

* **Intellij-Rust plugin version:** v0.2.0.2114-183
* **Rust toolchain version:** rustc 1.33.0-nightly (e2f221c75 2019-01-15)
* **IDE name and version:** CLion 2018.3
* **Operating system:** Ubuntu Cosmic

## Problem description
```rust
use std::mem::drop;

fn noreturn() -> ! {
    unsafe { libc::exit(228) }
}

struct Raii {}

impl Drop for Raii {
    fn drop(&mut self) {
        //do_dispose(self.x)
    }
}

fn consume(_r: Raii) {}

fn get_condition() -> bool {
    rand::random()
}

fn main() {
    let x = Raii {};

    if get_condition() {
        //A
        drop(x);
        noreturn();
    }
    consume(x); //got: Use of moved value
    //expected: x could only be dropped by in (A), but current location is unreachable from A
    //Rustc compiles this code with no warnings
}
```
